### PR TITLE
docs: AWS KMS

### PR DESCRIPTION
**Pull request by nklaassen**
Adding docs for AWS KMS support, which is being releasing in 15.0.0

### CODE REVIEW

File: docs/pages/choose-an-edition/teleport-enterprise/gcp-kms.mdx [hunk 1]
```diff
@@ -21,13 +21,10 @@ In short, private key material will never leave Google Cloud KMS.
...
 startup with no specific interventions required after configuration.
 For existing Teleport clusters that already have private key material on disk,
-a CA rotation must be performed. This ensures that the private key material
-only ever exists in KMS when this feature is enabled.
+a CA rotation must be performed.
 Read on to [migrating an existing cluster](#migrating-an-existing-cluster) to
 learn more.
 
-(!docs/pages/includes/cloud/call-to-action.mdx!)
-
```
Comment (nklaassen) [inline docs/pages/choose-an-edition/teleport-enterprise/gcp-kms.mdx:30]: This CTA reads "Teleport Team takes care of this setup for you so you can provide secure access to your infrastructure right away". I find that a bit disingenuous when Teleport Team doesn't use this feature at all, and getting the same CA key configuration as Team takes 0 extra steps

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx [hunk 2]
```diff
@@ -0,0 +1,187 @@
+---
+title: AWS KMS
+description: Configure Teleport to store CA private keys in the AWS Key Management Service
+h1: Store Teleport Private Keys in AWS KMS
+---
+
+This guide will show you how to set up your Teleport Cluster to use the AWS Key
+Management Service (KMS) to store and handle the CA private key material used to
+sign all certificates issued by your Teleport cluster.
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx:9]:
```suggestion
This guide will show you how to set up your Teleport cluster to use the AWS Key
Management Service (KMS) to store and handle the CA private key material used to
sign all certificates issued by your cluster.
```
- Capitalization
- Fix repetition

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+Teleport generates private key material for its internal Certificate Authorities
+(CAs) during the first Auth Server's initial startup.
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx:12]:
```suggestion
(CAs) during the first Auth Service instance's initial startup.
```
For consistency with the rest of the docs

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+You can create the policy from the AWS console, or via the AWS CLI.
+
+```code
+$ aws iam create-policy --policy-name TeleportKMS --policy-document file://policy.json
+```
+
+### Attach the AWS IAM policy
+
+Your Teleport Auth Server will need access to an IAM identity with this policy attached.
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx:106]:
```suggestion
Your Teleport Auth Service will need access to an IAM identity with this policy attached.
```

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+You can provide credentials via IAM Roles for Tasks, IAM Roles for EC2,
+credential files, or environment variables.
+See [Specifying Credentials](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials)
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx:109]: Would the `docs/pages/includes/aws-credentials.mdx` partial provide this information?

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+for details.
+
+If your Teleport Auth Service is running on an EC2 instance and already has an
+attached "IAM role for Amazon EC2", attach the above policy to the existing role.
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx:113]: What do the quotes mean here?

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+If the instance does not already have an attached role, create an IAM role with
+the above policy and attach it to your EC2 instance running the Teleport Auth
+Service.
+
+## Step 2/5. Configure your Auth Server to use KMS keys
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx:118]:
```suggestion
## Step 2/5. Configure your Auth Service to use KMS keys
```

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+CA key parameters are statically configured in the `teleport.yaml` configuration
+file of the Teleport Auth Services in your cluster.
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx:121]:
```suggestion
file of the Teleport Auth Service instances in your cluster.
```

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+In order for the Teleport Auth Service to generate new keys in AWS KMS and have
+them trusted by the rest of the cluster, you will need to rotate all of your
+Teleport CAs.
+
+`teleport start` will print a warning during startup if any CA needs to be rotated.
+Any users allowed the `update` verb for `cert_authority` resources in any of
+their Teleport roles will also see a cluster alert reminding them to rotate the
+CAs.
+
+CA rotation can be performed manually or semi-automatically, see our admin guide
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx:185]:
```suggestion
CA rotation can be performed manually or semi-automatically. See our admin guide
```

**Review by ptgott — COMMENTED** [review]
Mostly minor suggestions, with two questions.

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+Teleport cluster.
+When configured to use AWS KMS, all private key material for these CAs
+will be generated, stored, and used for signing inside of AWS KMS.
+Instead of the actual private key, Teleport will only store the ID of the KMS
+key.
+In short, private key material will never leave AWS KMS.
+
+If launching a new Teleport cluster this will all be handled during initial
+startup with no specific interventions required after configuration.
+For existing Teleport clusters that already have private key material on disk,
```
Comment (zmb3) [inline docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx:23]: I suspect most people interested in this feature will be those using Teleport on AWS with DynamoDB, so the "already on disk" point isn't entirely accurate..

**Review by zmb3 — APPROVED** [review]

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx (hunk 6 above)
Comment (nklaassen) [reply docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx]: this got replaced by the partial

File: docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx (hunk 5 above)
Comment (nklaassen) [reply docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx]: yeah, didn't know about that, thanks

**Review by ptgott — APPROVED** [review]

**Pull request merged**